### PR TITLE
Update User and Developer Guides.

### PR DESCRIPTION
**Pull request by pv101**

### CODE REVIEW

File: doc/src/user_guide.rst [hunk 1]
```diff
@@ -39,6 +39,16 @@ or greater. The backend requires:
 1. `CUDA <https://developer.nvidia.com/cuda-downloads>`_ >= 4.2
 2. `pycuda <http://mathema.tician.de/software/pycuda/>`_ >= 2011.2
 
+MIC Backend
+^^^^^^^^^^^
+
+The MIC backend targets Intel Xeon Phi co-processors. The backend
+requires:
+
+1. ICC >= 16.0
```
Comment (FreddieWitherden) [inline doc/src/user_guide.rst]: Do we need a version this recent?  I thought that Jin Seok was having trouble with recent releases?

File: doc/src/user_guide.rst [hunk 2]
```diff
@@ -181,6 +200,19 @@ Example::
...
     gimmik-max-nnz = 512
 
+[backend-mic]
+^^^^^^^^^^^^^^^^
+
+Parameterises the MIC backend with
+
+1. ``device-id`` --- for selecting which device(s) to run on:
+
+    *int* | *string* | ``local-rank``
```
Comment (FreddieWitherden) [inline doc/src/user_guide.rst]: Strings are not permitted.

File: doc/src/user_guide.rst (hunk 1 above)
Comment (Aerojspark) [inline doc/src/user_guide.rst]: I found that the recent intel MPSS 3.6 (system level software for Xeon Phi) has a problem to build pyMIC. MPSS 3.3, 3.4 or 3.5 don't have a problem. I will report this to pyMIC soon. I confirmed that the recent icc and mkl, from Intel(R) Composer XE 2013 SP1, 2015 or 2016, work well.

Comment (Aerojspark) [inline doc/src/user_guide.rst]: After reporting this issue, Michael fixed it and pyMIC can be compiled with MPSS 3.6.1. As pointed in pyMIC web page, I confirmed the following version of compiler and libraries (greater than Intel Composer XE 2013SP1 and MPSS 3.3) work well.
ICC >= 14.0
Intel MKL >= 11.1
Intel MPSS >= 3.3
pymic >= 0.7 (post commit 4d8a2da)

Comment (FreddieWitherden) [inline doc/src/user_guide.rst]: Thanks for following up on this.

@pv101 Can you update the request?

**Pull request merged**